pdr_backend/lake/merge_df.py

```diff
@@ -0,0 +1,98 @@
+from typing import List, Union
+
+from enforce_typing import enforce_types
+
+import polars as pl
+from pdr_backend.lake.plutil import set_col_values
+
+
+@enforce_types
+def merge_rawohlcv_dfs(rawohlcv_dfs: dict) -> pl.DataFrame:
+    """
+    @arguments
+      rawohlcv_dfs -- see class docstring
+
+    @return
+      mergedohlcv_df -- see class docstring
+    """
+    raw_dfs = rawohlcv_dfs
+
+    print("  Merge rawohlcv dataframes.")
+    merged_df = None
+    for exch_str in raw_dfs.keys():
+        for pair_str, raw_df in raw_dfs[exch_str].items():
+            assert "/" in pair_str, f"pair_str={pair_str} needs '/'"
+            assert "timestamp" in raw_df.columns
+
+            for raw_col in raw_df.columns:
+                if raw_col == "timestamp":
+                    continue
+                signal_str = raw_col  # eg "close"
+                merged_col = f"{exch_str}:{pair_str}:{signal_str}"
+                merged_df = _add_df_col(merged_df, merged_col, raw_df, raw_col)
+
+    merged_df = merged_df.select(_ordered_cols(merged_df.columns))  # type: ignore
+    _verify_df_cols(merged_df)
+    return merged_df
+
+
+@enforce_types
+def _add_df_col(
+    merged_df: Union[pl.DataFrame, None],
+    merged_col: str,  # eg "binance:BTC/USDT:close"
+    raw_df: pl.DataFrame,
+    raw_col: str,  # eg "close"
+) -> pl.DataFrame:
+    """
+    Does polars equivalent of: merged_df[merged_col] = raw_df[raw_col].
+    Tuned for this factory, by keeping "timestamp"
+    """
+    newraw_df = raw_df.with_columns(
+        pl.col(raw_col).alias(merged_col),
+    )
+    newraw_df = newraw_df.select(["timestamp", merged_col])
+
+    if merged_df is None:
+        merged_df = newraw_df
+    else:
+        merged_df = merged_df.join(newraw_df, on="timestamp", how="outer")
+        merged_df = merge_cols(merged_df, "timestamp", "timestamp_right")
```

Comment:
I believe the reason why we are getting _right columns, is because of the outer join. Generally, I don't like using outer joins as they can introduce unexpected data.

As long as we use left joins on expected timestamps, this should not happen. This can be done in two ways:
1. With a dataset you trust (i.e. kaiko that may have guaranteed timestamps)
2. With a dataset you build (i.e. just populate 1 column w/ expect values)

example
```
# Create a range of timestamps with 5-minute intervals
timestamps = pd.date_range(start='2023-01-01', periods=100, freq='5T')
timestamps = timestamps.astype(int) // 10**9  # Convert to Unix timestamps in seconds

# Create a Polars DataFrame
df = pl.DataFrame({
    "timestamp": timestamps
})

merged_df.extend(df)
merged_df = merged_df.join(newraw_df, on="timestamp", how="left")
```

This avoids:
- situations where you get _right/_left columns
- the need to run merge_cols()

This also enforces strict timestamp records in your data structure, rather than potentially introducing weird records due to outer join.

Reply:
Fyi I was building off the outer join code that you had introduced. 

We can't just use "data we trust". There's always going to be missing values etc and we need to account for it. Eg most people using pdr-backend will not be buying Kaiko data.

The example improvement you gave us to a good idea. If you want to make it happen, I recommend to make a new github issue.